app/services/bulk_upload/lettings/row_parser.rb

```diff
@@ -403,6 +422,29 @@ def setup_question?(question)
     log.form.setup_sections[0].subsections[0].questions.include?(question)
   end
 
+  def validate_if_log_already_exists
+    if log_already_exists?
+      error_message = "This is a duplicate log"
+
+      errors.add(:field_5, error_message) # location
```

Comment:
Is `field_5` correct here? In the bulk upload template it seems that `field_5` is the scheme, but the logic in the `location` method seems to find the location using this field.

Reply:
my understanding is old core doesn't have the concept of a location. so scheme here actually relates to new core location and `Management group code` is the scheme